sklearn/neighbors/dist_metrics.pyx

```diff
@@ -1111,7 +1100,13 @@ cdef class PyFuncDistance(DistanceMetric):
         with gil:
             x1arr = _buffer_to_ndarray(x1, size)
             x2arr = _buffer_to_ndarray(x2, size)
-            return self.func(x1arr, x2arr, **self.kwargs)
+            d = self.func(x1arr, x2arr, **self.kwargs)
+            try:
+                return d
+            except TypeError:
```

Comment:
Thanks @yenchenlin1994, that looks good. Would you like to submit a PR with that test? You'll have to put it in a function, probably in ``sklearn/neighbors/tests/test_dist_metrics.pyx``. You can use ``numpy.testing.assert_raises_regexp`` to assert that the expected exception is being raised. One more comment: to make the test faster, you could do much fewer than 1000x10 points: even something like 5x2 would probably do it.

Let me know if you need help putting that test together!